microservices/site-launch/lambda-functions/redirection-domain-validation/index.ts

```diff
@@ -55,34 +59,35 @@ export const redirectionDomainValidation = async (
         ref: DEFAULT_BRANCH,
       }
     )
-    if (response.status === 200) return // file already exist
-
-    if (response.status !== 404) {
-      const error = `Unexpected error occurred for redirection for ${primaryDomainSource} : 
-      ${JSON.stringify(response)}`
-      throw new Error(error)
+  } catch (error) {
+    if (
+      Object.prototype.hasOwnProperty.call(error, "status") &&
+      error.status === 404
```

Comment:
sure! 084d8e54
not sure about what you mean by throwing early here, we do throw in the `else` part right below when we get an **_unexpected_** error. note that a 404 file not found is an _**expected**_ error

Reply:
throw the exception early was what i was thinking; also, we can just `return` in the catch block if the error is 404 cos otherwise it'll always be true

Reply:
0cb2dcdd